upload/upload.go

```diff
@@ -21,7 +27,42 @@ var ArgHelp = `
 
 var Args = flag.NewFlagSet("upload", flag.ExitOnError)
 
-var config = Args.String("config", "", "S3 config file to use for uploading.")
+var configPath = Args.String("config", "", "S3 config file to use for uploading.")
+
+// Configuration struct for storing the s3cmd file values
+type Config struct {
+	AccessKey            string `ini:"access_key"`
+	SecretKey            string `ini:"secret_key"`
+	AccessToken          string `ini:"access_token"`
+	CheckSslCertificate  bool   `ini:"check_ssl_certificate"`
+	CheckSslHostname     bool   `ini:"check_ssl_hostname"`
+	SocketTimeout        int    `ini:"socket_timeout"`
+	Encoding             string `ini:"encoding"`
+	Encrypt              bool   `ini:"encrypt"`
+	HumanReadableSizes   bool   `ini:"human_readable_sizes"`
+	MultipartChunkSizeMb int64  `ini:"multipart_chunk_size_mb"`
+	HostBucket           string `ini:"host_bucket"`
+	HostBase             string `ini:"host_base"`
+	GuessMimeType        bool   `ini:"guess_mime_type"`
+	UseHttps             bool   `ini:"use_https"`
```

Comment:
most of these should have sane defaults

Reply:
I thought that it would be nice to keep all the configuration there, in case we want to change something in the future. I could though read in only the ones we are actually using for the uploading. What would be the best?

Reply:
Encrypt can be removed as that requires that the S3 backend has a crypto module.
give the rest production ready default values that is not backend specific. i.e. igore Hostbase and HostBucket

Reply:
MultipartChunkSizeMb is default 15Mb, but for us it needs to be bigger, 50-100Mb

Reply:
the booleans should default to `true`

Reply:
I think it makes sense to leave only the values that are needed: credentials, Hostbase/bucket and chunksize. And I will remove all the rest